models/hymod/include/Hymod.h

```diff
@@ -113,14 +113,13 @@ class hymod_kernel
             inital_mass_meters += current_state.Sr[i];
             final_mass_meters += next_state.Sr[i];
         }
-
-        // increase the inital mass by input value
-        inital_mass_meters += input_flux_meters;
+        // increase the inital mass by input value, NJF run function above adds input_flux to state before it is passed here
+        //inital_mass_meters += input_flux_meters;
```

Comment:
This feels problematic to me.  It might be worth discussing it a little more in depth, though I also don't want to tie this up in a can of worms if it's too hard to adjust the related upstream behavior in the `run` function.

Regardless, if this is going to be commented out as such, shouldn't we also remove the `input_flux_meters` function parameter?

Reply:
What is problematic about it?  line 56 clearly adds the `input_flux` to the state, so doing it here double counts that input and causes the mass balance to fail incorrectly.  The only other solution would be to copy the state at the beginning of the `run` function so the mass balance check has a "clean" state to check against.

I do agree if we don't need the parameter, it should be dropped from the calling statement.

Reply:
Having the _input_ flux not be separated from the _initial_ state stands out in my mind for some reason, like the type of thing that could have unanticipated, subtle implications.  On the other hand, I may just be biased from having seen it done that way previously, and from seeing the fluxes separate from the _next_ state.  

This is really a question about something beyond the scope of this PR, though, so it can be tabled if it needs to be pursued.  For now, I think as long as the parameter is removed, this should be fine.

Reply:
Feel free to open an issue outlining your other concern and we can circle back to it.

Reply:
paramter removed in ac00d07